python/src/array.cpp

```diff
@@ -113,14 +106,97 @@ void fill_vector(T list, std::vector<U>& vals) {
       fill_vector(l.template cast<py::list>(), vals);
     } else if (py::isinstance<py::tuple>(*list.begin())) {
       fill_vector(l.template cast<py::tuple>(), vals);
+    } else if (py::isinstance<array>(l)) {
+      auto arr = py::cast<array>(l);
+      arr.eval();
```

Comment:
Need to call `eval()` here since `arr` may come from transformations so that it does not hold the actual raw data.